appeals/api/src/server/mappers/mapper-factory.js

```diff
@@ -0,0 +1,257 @@
+import { apiMappers } from './api/index.js';
+import { contextEnum } from './context-enum.js';
+import { APPEAL_CASE_STAGE, APPEAL_CASE_TYPE, APPEAL_DOCUMENT_TYPE } from 'pins-data-model';
+
+/** @typedef {import('@pins/appeals.api').Schema.Appeal} Appeal */
+/** @typedef {import('@pins/appeals.api').Schema.AppealType} AppealType */
+/** @typedef {import('@pins/appeals.api').Api.Appeal} AppealDTO */
+/** @typedef {import('@pins/appeals.api').Api.Folder} Folder */
+/** @typedef {{ appeal: Appeal, appealTypes: AppealType[]|undefined, context: keyof contextEnum|undefined }} MappingRequest */
+
+/**
+ *
+ * @param {MappingRequest} mappingRequest
+ * @returns {AppealDTO|undefined}
+ */
+export const mapCase = ({ appeal, appealTypes = [], context = contextEnum.appealDetails }) => {
+	if (context && appeal?.id && appeal?.caseCreatedDate) {
+		const caseMap = createCaseMap({ appeal, appealTypes, context });
+		return createCaseLayout(caseMap, { appeal, appealTypes, context });
+	}
+};
+
+/**
+ *
+ * @param {MappingRequest} mappingRequest
+ * @returns {Map<string, *>}
+ */
+function createCaseMap(mappingRequest) {
+	const { appeal } = mappingRequest;
+
+	const casedata = createMap(apiMappers.apiSharedMappers, mappingRequest);
+	switch (appeal.appealType?.key) {
+		case APPEAL_CASE_TYPE.W: {
+			const s78 = createMap(apiMappers.apiS78Mappers, mappingRequest);
+			updateMap(casedata, s78);
+			break;
+		}
+		case APPEAL_CASE_TYPE.D:
+		default: {
+			const has = createMap(apiMappers.apiHasMappers, mappingRequest);
+			updateMap(casedata, has);
+			break;
+		}
+	}
+
+	return casedata;
+}
+
+/**
+ *
+ * @param {Object<string, *>} mappers
+ * @param {MappingRequest} mappingRequest
+ * @returns {Map<string, *>}
+ */
+function createMap(mappers, mappingRequest) {
+	const casedata = new Map([]);
+	Object.entries(mappers).forEach(([key, mapper]) => {
+		const mapped = mapper(mappingRequest);
+		if (mapped) {
+			casedata.set(key, mapped);
+		}
+	});
+	return casedata;
+}
+
+/**
+ *
+ * @param {Map<string, *>} existingMap
+ * @param {Map<string, *>} newMap
+ */
+function updateMap(existingMap, newMap) {
+	Array.from(existingMap.entries()).forEach(([key, value]) => {
+		const newValue = newMap.get(key);
+		if (newValue) {
+			existingMap.set(key, {
+				...value,
+				...newValue
+			});
+		}
+	});
+}
+
+/**
+ *
+ * @param {Map<string, *>} caseMap
+ * @param {MappingRequest} mappingRequest
+ */
+function createCaseLayout(caseMap, mappingRequest) {
```

Comment:
it's a pedantic point, but it might be better not to include `case` in the name of this function, as it seems to be mapping other things besides the case itself, depending on the input - I'm not sure if there's a generic term we're using to describe these pages, but maybe something like `createDataLayout` or `createFormattedData`?